src/Features/Core/Completion/MostRecentlyUsedList.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal class MostRecentlyUsedList<T>
+    {
+        private readonly object gate = new object();
+        private readonly List<T> items = new List<T>();
+        private readonly int maxSize;
+
+        public MostRecentlyUsedList(int maxSize = 10)
+        {
+            this.maxSize = maxSize;
+        }
+
+        public void Add(T value)
+        {
+            lock (gate)
+            {
+                var removed = items.Remove(value);
```

Comment:
Are we OK using the EqualityComparer<T>.Default within this and other 'searchy' List<T> operations?